src/main/java/org/apache/mesos/scheduler/plan/BlockFactory.java

```diff
@@ -11,6 +12,6 @@
  * detected for a Task.
  */
 public interface BlockFactory {
-    Block getBlock(String taskType, TaskSpecification taskSpecification)
-            throws InvalidRequirementException;
+    Block getBlock(TaskSpecification taskSpecification) throws Block.InvalidException;
+    List<Block> getBlocks(List<TaskSpecification> taskSpecifications) throws Block.InvalidException;
```

Comment:
See here: https://github.com/mesosphere/dcos-commons/pull/220/files#diff-b30db19793859d5a98b8fddd400047eeR145

Are you asking about the method addition or the Exception?